runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -1787,6 +1792,63 @@ TR::Register *J9::X86::TreeEvaluator::multianewArrayEvaluator(TR::Node *node, TR
 
    node->setRegister(targetReg);
    return targetReg;
+}
+
+/**
+ * Generate code for multianewarray
+ *
+ * Checks the number of dimensions. For 1 dimensional arrays call the helper, for >1 call
+ * generateMultianewArrayWithInlineAllocators.
+*/
+TR::Register *J9::X86::TreeEvaluator::multianewArrayEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+
+   TR::Compilation *comp = cg->comp();
+   TR_ASSERT_FATAL(comp->target().is64Bit(), "multianewArrayEvaluator is only supported on 64-bit JVMs!");
+
+   bool useInlineAllocator = false;
+   TR::Node *secondChild = node->getSecondChild();    // Number of dimensions - this is fixed in the bytecode, so compile time constant
+
+   // if secondChild is an iconst we can read the number of dimensions. Only generate inline code if nDims > 1
+   if (secondChild->getOpCodeValue() == TR::iconst)
```

Comment:
This can't be anything other than a constant.  You should make this an assert instead and get rid of the `else` path.